src/hooks/useKanban.js

```diff
@@ -0,0 +1,317 @@
+    try {
+      const voterId = user ? user.uid : getVisitorId();
+      const column = Object.values(columns).find(col => 
+        col.tasks?.some(task => task.id === taskId)
+      );
+      
+      if (!column) return false;
+      
+      const task = column.tasks.find(t => t.id === taskId);
+      return task?.votedBy?.includes(voterId) || false;
+    } catch (error) {
+      console.error('Erro ao verificar voto:', error);
+      return false;
+    }
+  };
+
+  // Gera um ID único para visitantes não logados
+  const getVisitorId = () => {
+    let visitorId = localStorage.getItem('visitorId');
+    if (!visitorId) {
+visitorId = 'visitor_' + Date.now() + '_' + Math.random().toString(36).substr(2, 9);
+      localStorage.setItem('visitorId', visitorId);
+    }
+    return visitorId;
+  };
+
+  // Move uma task entre colunas
+  const moveTask = async (sourceColumnId, destinationColumnId, taskId, sourceIndex, destinationIndex) => {
+    if (!user) return;
+    
+    try {
+      const sourceColumn = columns[sourceColumnId];
+      const destinationColumn = columns[destinationColumnId];
+      
+      if (!sourceColumn || !destinationColumn) return;
+
+      const sourceTasks = [...(sourceColumn.tasks || [])];
+      const [movedTask] = sourceTasks.splice(sourceIndex, 1);
+
+      const destinationTasks = sourceColumnId === destinationColumnId 
+        ? sourceTasks 
+        : [...(destinationColumn.tasks || [])];
+
+      destinationTasks.splice(destinationIndex, 0, movedTask);
+
+      if (sourceColumnId === destinationColumnId) {
+        await updateColumn(sourceColumnId, { tasks: destinationTasks });
+      } else {
+        await Promise.all([
+          updateColumn(sourceColumnId, { tasks: sourceTasks }),
+          updateColumn(destinationColumnId, { tasks: destinationTasks })
+        ]);
+      }
+    } catch (error) {
+      console.error('Erro ao mover task:', error);
+    }
+  };
+
+  // Atualiza uma task existente
+  const updateTask = async (columnId, taskId, updates) => {
+    if (!user) return;
+    
+    try {
+      const column = columns[columnId];
+      if (!column) return;
+
+      const taskIndex = column.tasks.findIndex(t => t.id === taskId);
+      if (taskIndex === -1) return;
+
+      const updatedTask = {
+        ...column.tasks[taskIndex],
+        ...updates,
+        updatedAt: new Date(),
+        updatedBy: user.uid
+      };
+
+      const updatedTasks = [...column.tasks];
+      updatedTasks[taskIndex] = updatedTask;
+
+      await updateColumn(columnId, { tasks: updatedTasks });
+    } catch (error) {
+      console.error('Erro ao atualizar task:', error);
+    }
+  };
+
+  // Remove uma task
+  const deleteTask = async (columnId, taskId) => {
+    if (!user) return;
+    
+    try {
+      const column = columns[columnId];
+      if (!column) return;
+
+      const updatedTasks = column.tasks.filter(task => task.id !== taskId);
+      await updateColumn(columnId, { tasks: updatedTasks });
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const updatedTasks = column.tasks.filter(task => task.id !== taskId);
setColumns(prev => ({
  ...prev,
  [columnId]: { ...prev[columnId], tasks: updatedTasks }
}));
try {
  await updateColumn(columnId, { tasks: updatedTasks });
} catch (error) {
  // Revert on failure
  setColumns(prev => ({ ...prev, [columnId]: column }));
  throw error;
}
```
Consider adding optimistic updates to improve user experience by updating the UI before the API call completes.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;